Assets/MRTK/SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -1513,11 +1513,13 @@ public void OnSpeechKeywordRecognized(SpeechEventData eventData)
 
             if (eventData.Command.Keyword == VoiceCommand && (!VoiceRequiresFocus || HasFocus) && CanInteract())
             {
+                bool hadFocus = HasFocus;
```

Comment:
Thanks for this PR! After looking through the fix a bit, I think I've identified a different potential problem area that might solve this issue:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/1a9fb98dd648cd4e8a46b65b805f6c9ef690cdea/Assets/MRTK/SDK/Features/UX/Interactable/Scripts/Interactable.cs#L1300-L1313

I think the `if` checks are backwards. Why check if `HasFocus` is false just to set it to false again? This method is started as a coroutine at the end of `StartGlobalVisual`, which is called in this code block

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/1a9fb98dd648cd4e8a46b65b805f6c9ef690cdea/Assets/MRTK/SDK/Features/UX/Interactable/Scripts/Interactable.cs#L1294

Reply:
Though...thinking this through a little more, I think the underlying issue is that the rework in #6104 removed the distinction between "having focus" and "being in a focused visual state", which this code path previously relied on. Changing the lines I posted above would introduce an edge case where if I'm both focusing a button with a pointing ray and using a voice command to trigger it, the visual state would be reset even though my pointing ray is still focusing it.

Could work around this with something like `if (focusingPointers.Count == 0)`, but then I'm not sure what to do with `HasPress`

Reply:
What do you think?

Reply:
I looked into it and I agree that it would be a cleaner solution to fix this in the `GlobalVisualReset`. 

I tested fixing the backwards `HasFocus`. I believe the `HasPress` is not linked to this specific issue, so I left it as it is to not unnecessarily break anything else.

Changing it to the following, seems to fix the issue:
```csharp
if (HasFocus && focusingPointers.Count == 0)
{
      HasFocus = false;
}
```

I pushed a change implementing this.

Reply:
Thank you! Looks great